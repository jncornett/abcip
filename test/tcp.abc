c( a.data="Groovy Baby!", b.data="shagalicious" )
c( ip4:a="192.168.1.2", b="10.10.4.182"; tcp:a=8420, b=80 )

a(syn)
b(syn, ack)
a(ack)

a(ack, len=16)
b(ack, len=16)

a(fin,ack)
b(fin,ack)
a(ack)

c(a=48621)
a(syn)
b(syn, ack)
a(ack)

a(ack, len=16)

a(fin,ack)
b(fin,ack)
a(ack)

