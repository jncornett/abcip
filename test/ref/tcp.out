IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.8420 > 10.10.4.182.80: Flags [S], cksum 0x9e43 (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 aa65 c0a8 0102 0a0a  .(....@..e......
	0x0020:  04b6 20e4 0050 0000 0001 0000 0000 5002  .....P........P.
	0x0030:  2000 9e43 0000                           ...C..
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.10.4.182.80 > 192.168.1.2.8420: Flags [S.], cksum 0x9e31 (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 aa65 0a0a 04b6 c0a8  .(....@..e......
	0x0020:  0102 0050 20e4 0000 0001 0000 0002 5012  ...P..........P.
	0x0030:  2000 9e31 0000                           ...1..
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.8420 > 10.10.4.182.80: Flags [.], cksum 0x9e32 (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 aa64 c0a8 0102 0a0a  .(....@..d......
	0x0020:  04b6 20e4 0050 0000 0002 0000 0002 5010  .....P........P.
	0x0030:  2000 9e32 0000                           ...2..
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 56)
    192.168.1.2.8420 > 10.10.4.182.80: Flags [.], cksum 0xbf1f (correct), seq 1:17, ack 1, win 8192, length 16
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0038 0003 0000 4006 aa53 c0a8 0102 0a0a  .8....@..S......
	0x0020:  04b6 20e4 0050 0000 0002 0000 0002 5010  .....P........P.
	0x0030:  2000 bf1f 0000 4772 6f6f 7679 2042 6162  ......Groovy.Bab
	0x0040:  7921 4772 6f6f                           y!Groo
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 56)
    10.10.4.182.80 > 192.168.1.2.8420: Flags [.], cksum 0x4ac0 (correct), seq 1:17, ack 17, win 8192, length 16
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0038 0002 0000 4006 aa54 0a0a 04b6 c0a8  .8....@..T......
	0x0020:  0102 0050 20e4 0000 0002 0000 0012 5010  ...P..........P.
	0x0030:  2000 4ac0 0000 7368 6167 616c 6963 696f  ..J...shagalicio
	0x0040:  7573 7368 6167                           usshag
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.8420 > 10.10.4.182.80: Flags [F.], cksum 0x9e11 (correct), seq 17, ack 17, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0004 0000 4006 aa62 c0a8 0102 0a0a  .(....@..b......
	0x0020:  04b6 20e4 0050 0000 0012 0000 0012 5011  .....P........P.
	0x0030:  2000 9e11 0000                           ......
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 40)
    10.10.4.182.80 > 192.168.1.2.8420: Flags [F.], cksum 0x9e10 (correct), seq 17, ack 18, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0003 0000 4006 aa63 0a0a 04b6 c0a8  .(....@..c......
	0x0020:  0102 0050 20e4 0000 0012 0000 0013 5011  ...P..........P.
	0x0030:  2000 9e10 0000                           ......
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.8420 > 10.10.4.182.80: Flags [.], cksum 0x9e10 (correct), ack 18, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0005 0000 4006 aa61 c0a8 0102 0a0a  .(....@..a......
	0x0020:  04b6 20e4 0050 0000 0013 0000 0013 5010  .....P........P.
	0x0030:  2000 9e10 0000                           ......
IP (tos 0x0, ttl 64, id 6, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.48621 > 10.10.4.182.80: Flags [S], cksum 0x013a (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0006 0000 4006 aa60 c0a8 0102 0a0a  .(....@..`......
	0x0020:  04b6 bded 0050 0000 0001 0000 0000 5002  .....P........P.
	0x0030:  2000 013a 0000                           ...:..
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 40)
    10.10.4.182.80 > 192.168.1.2.48621: Flags [S.], cksum 0x0128 (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0004 0000 4006 aa62 0a0a 04b6 c0a8  .(....@..b......
	0x0020:  0102 0050 bded 0000 0001 0000 0002 5012  ...P..........P.
	0x0030:  2000 0128 0000                           ...(..
IP (tos 0x0, ttl 64, id 7, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.48621 > 10.10.4.182.80: Flags [.], cksum 0x0129 (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0007 0000 4006 aa5f c0a8 0102 0a0a  .(....@.._......
	0x0020:  04b6 bded 0050 0000 0002 0000 0002 5010  .....P........P.
	0x0030:  2000 0129 0000                           ...)..
IP (tos 0x0, ttl 64, id 8, offset 0, flags [none], proto TCP (6), length 56)
    192.168.1.2.48621 > 10.10.4.182.80: Flags [.], cksum 0x423c (correct), seq 1:17, ack 1, win 8192, length 16
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0038 0008 0000 4006 aa4e c0a8 0102 0a0a  .8....@..N......
	0x0020:  04b6 bded 0050 0000 0002 0000 0002 5010  .....P........P.
	0x0030:  2000 423c 0000 7679 2042 6162 7921 4772  ..B<..vy.Baby!Gr
	0x0040:  6f6f 7679 2042                           oovy.B
IP (tos 0x0, ttl 64, id 9, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.48621 > 10.10.4.182.80: Flags [F.], cksum 0x0118 (correct), seq 17, ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0009 0000 4006 aa5d c0a8 0102 0a0a  .(....@..]......
	0x0020:  04b6 bded 0050 0000 0012 0000 0002 5011  .....P........P.
	0x0030:  2000 0118 0000                           ......
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 40)
    10.10.4.182.80 > 192.168.1.2.48621: Flags [F.], cksum 0x0117 (correct), seq 1, ack 18, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0005 0000 4006 aa61 0a0a 04b6 c0a8  .(....@..a......
	0x0020:  0102 0050 bded 0000 0002 0000 0013 5011  ...P..........P.
	0x0030:  2000 0117 0000                           ......
IP (tos 0x0, ttl 64, id 10, offset 0, flags [none], proto TCP (6), length 40)
    192.168.1.2.48621 > 10.10.4.182.80: Flags [.], cksum 0x0117 (correct), ack 2, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 000a 0000 4006 aa5c c0a8 0102 0a0a  .(....@..\......
	0x0020:  04b6 bded 0050 0000 0013 0000 0003 5010  .....P........P.
	0x0030:  2000 0117 0000                           ......
